# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/clang/gcc-path.patch
# Copyright (C) 2021 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- clang-20.1.0.src/lib/Driver/ToolChains/Gnu.cpp.orig	2025-03-04 20:51:43.000000000 +0100
+++ clang-20.1.0.src/lib/Driver/ToolChains/Gnu.cpp	2025-03-05 15:16:14.757511107 +0100
@@ -2417,43 +2417,38 @@
   // lists should shrink over time. Please don't add more elements to *Triples.
   static const char *const AArch64LibDirs[] = {"/lib64", "/lib"};
   static const char *const AArch64Triples[] = {
-      "aarch64-none-linux-gnu", "aarch64-redhat-linux", "aarch64-suse-linux"};
+	"aarch64-t2-linux-gnu", "aarch64-t2-linux-musl"};
   static const char *const AArch64beLibDirs[] = {"/lib"};
-  static const char *const AArch64beTriples[] = {"aarch64_be-none-linux-gnu"};
+  static const char *const AArch64beTriples[] = {"aarch64_be-t2-linux-gnu"};
 
   static const char *const ARMLibDirs[] = {"/lib"};
-  static const char *const ARMTriples[] = {"arm-linux-gnueabi"};
-  static const char *const ARMHFTriples[] = {"arm-linux-gnueabihf",
-                                             "armv7hl-redhat-linux-gnueabi",
-                                             "armv6hl-suse-linux-gnueabi",
-                                             "armv7hl-suse-linux-gnueabi"};
+  static const char *const ARMTriples[] = {"arm-t2-linux-gnueabi"};
+  static const char *const ARMHFTriples[] = {"arm-t2-linux-gnueabihf",
+                                             "armv6hl-t2-linux-gnueabi",
+                                             "armv7hl-t2-linux-gnueabi"};
   static const char *const ARMebLibDirs[] = {"/lib"};
-  static const char *const ARMebTriples[] = {"armeb-linux-gnueabi"};
+  static const char *const ARMebTriples[] = {"armeb-t2-linux-gnueabi"};
   static const char *const ARMebHFTriples[] = {
-      "armeb-linux-gnueabihf", "armebv7hl-redhat-linux-gnueabi"};
+      "armeb-t2-linux-gnueabihf", "armebv7hl-t2t-linux-gnueabi"};
 
   static const char *const AVRLibDirs[] = {"/lib"};
   static const char *const AVRTriples[] = {"avr"};
 
   static const char *const CSKYLibDirs[] = {"/lib"};
   static const char *const CSKYTriples[] = {
-      "csky-linux-gnuabiv2", "csky-linux-uclibcabiv2", "csky-elf-noneabiv2"};
+      "csky-g2-linux-gnuabiv2"};
 
   static const char *const X86_64LibDirs[] = {"/lib64", "/lib"};
   static const char *const X86_64Triples[] = {
-      "x86_64-linux-gnu",       "x86_64-unknown-linux-gnu",
-      "x86_64-pc-linux-gnu",    "x86_64-redhat-linux6E",
-      "x86_64-redhat-linux",    "x86_64-suse-linux",
-      "x86_64-manbo-linux-gnu", "x86_64-slackware-linux",
-      "x86_64-unknown-linux",   "x86_64-amazon-linux"};
-  static const char *const X32Triples[] = {"x86_64-linux-gnux32",
-                                           "x86_64-pc-linux-gnux32"};
+      "x86_64-t2-linux-gnu",    "x86_64-t2-linux-musl"};
+  static const char *const X32Triples[] = {"x86_64-t2-linux-gnux32"};
   static const char *const X32LibDirs[] = {"/libx32", "/lib"};
   static const char *const X86LibDirs[] = {"/lib32", "/lib"};
   static const char *const X86Triples[] = {
-      "i586-linux-gnu",      "i686-linux-gnu",        "i686-pc-linux-gnu",
-      "i386-redhat-linux6E", "i686-redhat-linux",     "i386-redhat-linux",
-      "i586-suse-linux",     "i686-montavista-linux",
+      "i386-t2-linux-gnu",  "i486-t2-linux-gnu",  "i586-t2-linux-gnu",
+      "i686-t2-linux-gnu",
+      "i386-t2-linux-musl", "i486-t2-linux-musl", "i586-t2-linux-musl",
+      "i686-t2-linux-musl"
   };
 
   static const char *const LoongArch64LibDirs[] = {"/lib64", "/lib"};
@@ -2466,27 +2461,23 @@
 
   static const char *const MIPSLibDirs[] = {"/libo32", "/lib"};
   static const char *const MIPSTriples[] = {
-      "mips-linux-gnu", "mips-mti-linux", "mips-mti-linux-gnu",
-      "mips-img-linux-gnu", "mipsisa32r6-linux-gnu"};
+      "mips-t2-linux-gnu", "mipsisa32r6-t2-linux-gnu"};
   static const char *const MIPSELLibDirs[] = {"/libo32", "/lib"};
-  static const char *const MIPSELTriples[] = {"mipsel-linux-gnu",
-                                              "mips-img-linux-gnu"};
+  static const char *const MIPSELTriples[] = {"mipsel-t2-linux-gnu", "mips-t2-linux-musl"};
 
   static const char *const MIPS64LibDirs[] = {"/lib64", "/lib"};
   static const char *const MIPS64Triples[] = {
-      "mips-mti-linux-gnu", "mips-img-linux-gnu", "mips64-linux-gnuabi64",
-      "mipsisa64r6-linux-gnu", "mipsisa64r6-linux-gnuabi64"};
+      "mips64-t2-linux-gnuabi64", "mips64-t2-linux-muslbi64"};
   static const char *const MIPS64ELLibDirs[] = {"/lib64", "/lib"};
   static const char *const MIPS64ELTriples[] = {
-      "mips-mti-linux-gnu", "mips-img-linux-gnu", "mips64el-linux-gnuabi64",
-      "mipsisa64r6el-linux-gnu", "mipsisa64r6el-linux-gnuabi64"};
+      "mips64el-t2-linux-gnuabi64", "mips64el-linux-muslabi64"};
 
   static const char *const MIPSN32LibDirs[] = {"/lib32"};
-  static const char *const MIPSN32Triples[] = {"mips64-linux-gnuabin32",
-                                               "mipsisa64r6-linux-gnuabin32"};
+  static const char *const MIPSN32Triples[] = {"mips64-t2-linux-gnuabin32",
+                                               "mips64-t2-linux-muslabin32"};
   static const char *const MIPSN32ELLibDirs[] = {"/lib32"};
   static const char *const MIPSN32ELTriples[] = {
-      "mips64el-linux-gnuabin32", "mipsisa64r6el-linux-gnuabin32"};
+      "mips64el-t2-linux-gnuabin32", "mips64el-t2-linux-muslabin32"};
 
   static const char *const MSP430LibDirs[] = {"/lib"};
   static const char *const MSP430Triples[] = {"msp430-elf"};
@@ -2496,38 +2487,34 @@
       "powerpc-unknown-linux-gnu",
       // On 32-bit PowerPC systems running SUSE Linux, gcc is configured as a
       // 64-bit compiler which defaults to "-m32", hence "powerpc64-suse-linux".
-      "powerpc64-suse-linux", "powerpc-montavista-linuxspe"};
+      "powerpc64-t2-linux-gnu", "powerpc64-t2-linux-musl"};
   static const char *const PPCLELibDirs[] = {"/lib32", "/lib"};
-  static const char *const PPCLETriples[] = {"powerpcle-unknown-linux-gnu",
-                                             "powerpcle-linux-musl"};
+  static const char *const PPCLETriples[] = {"powerpcle-t2-linux-gnu",
+                                             "powerpcle-t2-linux-musl"};
 
   static const char *const PPC64LibDirs[] = {"/lib64", "/lib"};
-  static const char *const PPC64Triples[] = {"powerpc64-unknown-linux-gnu",
-                                             "powerpc64-suse-linux",
-                                             "ppc64-redhat-linux"};
+  static const char *const PPC64Triples[] = {"powerpc64-t2-linux-gnu",
+                                             "powerpc64-t2-linux-musl"};
   static const char *const PPC64LELibDirs[] = {"/lib64", "/lib"};
   static const char *const PPC64LETriples[] = {
-      "powerpc64le-unknown-linux-gnu", "powerpc64le-none-linux-gnu",
-      "powerpc64le-suse-linux", "ppc64le-redhat-linux"};
+      "powerpc64le-t2-linux-gnu", "powerpc64le-t2e-linux-musl"};
 
   static const char *const RISCV32LibDirs[] = {"/lib32", "/lib"};
   static const char *const RISCV32Triples[] = {"riscv32-unknown-linux-gnu",
                                                "riscv32-unknown-elf"};
   static const char *const RISCV64LibDirs[] = {"/lib64", "/lib"};
-  static const char *const RISCV64Triples[] = {"riscv64-unknown-linux-gnu",
-                                               "riscv64-unknown-elf"};
+  static const char *const RISCV64Triples[] = {"riscv64-t2-linux-gnu", "riscv64-t2-linux-musl"};
 
   static const char *const SPARCv8LibDirs[] = {"/lib32", "/lib"};
-  static const char *const SPARCv8Triples[] = {"sparc-linux-gnu",
-                                               "sparcv8-linux-gnu"};
+  static const char *const SPARCv8Triples[] = {
+                                               "sparcv8-t2-linux-gnu",
+                                                "sparcv9-t2-linux-gnu"};
   static const char *const SPARCv9LibDirs[] = {"/lib64", "/lib"};
-  static const char *const SPARCv9Triples[] = {"sparc64-linux-gnu",
-                                               "sparcv9-linux-gnu"};
+  static const char *const SPARCv9Triples[] = {"sparc64-t2-linux-gnu"};
 
   static const char *const SystemZLibDirs[] = {"/lib64", "/lib"};
   static const char *const SystemZTriples[] = {
-      "s390x-unknown-linux-gnu", "s390x-ibm-linux-gnu", "s390x-suse-linux",
-      "s390x-redhat-linux"};
+      "s390x-t2-linux-gnu", "s390x-t2-linux-musl"};
 
   using std::begin;
   using std::end;
--- clang-13.0.0.src/lib/Driver/ToolChains/PPCLinux.cpp.vanilla	2021-12-21 23:19:38.231936574 +0100
+++ clang-13.0.0.src/lib/Driver/ToolChains/PPCLinux.cpp	2021-12-21 23:20:38.411939678 +0100
@@ -16,7 +16,7 @@
 
 void PPCLinuxToolChain::AddClangSystemIncludeArgs(const ArgList &DriverArgs,
                                                   ArgStringList &CC1Args) const {
-  if (!DriverArgs.hasArg(clang::driver::options::OPT_nostdinc) &&
+  if (false && !DriverArgs.hasArg(clang::driver::options::OPT_nostdinc) &&
       !DriverArgs.hasArg(options::OPT_nobuiltininc)) {
     const Driver &D = getDriver();
     SmallString<128> P(D.ResourceDir);
